conda_smithy/configure_feedstock.py

```diff
@@ -59,112 +22,292 @@
+            fast_finish_text += "del {fast_finish_script}.py"
 
-        build_setup = ""
+        fast_finish_text = fast_finish_text.format(
+            get_fast_finish_script=get_fast_finish_script,
+            fast_finish_script=fast_finish_script,
+        )
 
+        fast_finish_text = fast_finish_text.strip()
+        fast_finish_text = fast_finish_text.replace("\n", "\n        ")
+    else:
         # If the recipe supplies its own conda-forge-build-setup script,
         # we use it instead of the global one.
-        cfbs_fpath = os.path.join(forge_dir, 'recipe',
-                                  'run_conda_forge_build_setup_linux')
         if os.path.exists(cfbs_fpath):
-            build_setup += textwrap.dedent("""\
-                # Overriding global conda-forge-build-setup with local copy.
-                source /recipe_root/run_conda_forge_build_setup_linux
-
-            """)
+            get_fast_finish_script += "cat {recipe_dir}/ff_ci_pr_build.py".format(
+                recipe_dir=forge_config["recipe_dir"])
         else:
-            build_setup += textwrap.dedent("""\
-                source run_conda_forge_build_setup
+            get_fast_finish_script += "curl https://raw.githubusercontent.com/conda-forge/conda-forge-build-setup-feedstock/master/recipe/ff_ci_pr_build.py"  # NOQA
 
-            """)
+        fast_finish_text = fast_finish_text.format(
+            get_fast_finish_script=get_fast_finish_script
+        )
 
-        # If there is a "yum_requirements.txt" file in the recipe, we honour it.
-        yum_requirements_fpath = os.path.join(forge_dir, 'recipe',
-                                              'yum_requirements.txt')
-        if os.path.exists(yum_requirements_fpath):
-            with open(yum_requirements_fpath) as fh:
-                requirements = [line.strip() for line in fh
-                                if line.strip() and not line.strip().startswith('#')]
-            if not requirements:
-                raise ValueError("No yum requirements enabled in the "
-                                 "yum_requirements.txt, please remove the file "
-                                 "or add some.")
-            build_setup += textwrap.dedent("""\
+        fast_finish_text = fast_finish_text.strip()
+    return fast_finish_text
 
-                # Install the yum requirements defined canonically in the
-                # "recipe/yum_requirements.txt" file. After updating that file,
-                # run "conda smithy rerender" and this line be updated
-                # automatically.
-                /usr/bin/sudo -n yum install -y {}
 
+def _render_ci_provider(provider_name, jinja_env, forge_config, forge_dir, platform, arch,
+                        fast_finish_text, platform_target_path, platform_template_file,
+                        platform_specific_setup, keep_noarch=False, extra_platform_files=None):
+    metas = conda_build.api.render(os.path.join(forge_dir, 'recipe'),
```

Comment:
@msarahan, shouldn't the channel urls be sent to the render method?

Reply:
they're in forge_config?  Yes, they should be passed in here.  Please confirm where they are, and I'll add them in.

Reply:
It's in `forge_config['channels']['sources']`

Reply:
It's a list with highest order at the beginning